### PR TITLE
Clarify last element in str.{r,}splitn documentation

### PR DESCRIPTION
**Pull request by angelsl**
An attempt at #36202.

I'm not sure if my wording is actually clearer, to be honest...

### CODE REVIEW

File: src/libcollections/str.rs [hunk 1]
```diff
@@ -1055,8 +1055,8 @@ impl str {
     /// An iterator over substrings of the given string slice, separated by a
     /// pattern, restricted to returning at most `count` items.
     ///
-    /// The last element returned, if any, will contain the remainder of the
```
Comment (frewsxcv) [inline src/libcollections/str.rs]: I think, as a whole, your changes are an improvement and make it easier to understand this concept.

> If a `count`th substring

I personally stumbled on this though. It took me a few re-reads to understand this is talking about "substring with `count`th number of items". I can't think of a better suggestion at the moment without making it too verbose. Going to think about this...

Comment (GuillaumeGomez) [inline src/libcollections/str.rs]: For non-english reader, this is particurlarly hard.

Comment (frewsxcv) [inline src/libcollections/str.rs]: I think that's better. Or maybe even:

> If `count` substrings are returned, the last substring (the `count`th substring) will contain the remainder of the string.

What do you think @GuillaumeGomez ?

Comment (GuillaumeGomez) [inline src/libcollections/str.rs]: @frewsxcv: I prefer @durka's proposition. :)

Comment (frewsxcv) [inline src/libcollections/str.rs]: I'm generally against single letter variable names, but considering the method name is `splitn`, I think `n` makes sense. What do you think @angelsl? If you're alright with it, can you rename the parameter and update the instances of `count` in the doc comment?

Comment (angelsl) [inline src/libcollections/str.rs]: Sounds good to me. I've amended the commit.

**Comment by frewsxcv** [comment]
Looks great, thanks! r=me

**Pull request merged**